pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl

```diff
@@ -0,0 +1,351 @@
+version 1.0
+
+import "UMIAwareDuplicateMarking.wdl" as UmiMD
+
+workflow RNAWithUMIsPipeline {
+	input {
+		File bam
+		String read1Structure
+		String read2Structure
+		File starIndex
+		String output_basename
+		File gtf
+
+		File ref
+		File refIndex
+		File refDict
+		File refFlat
+		File ribosomalIntervals
+		File exonBedFile
+	}
+
+	call ExtractUMIs {
+		input:
+			bam = bam,
+			read1Structure = read1Structure,
+			read2Structure = read2Structure
+	}
+
+	call STAR {
+		input:
+			bam = ExtractUMIs.bam_umis_extracted,
+			starIndex = starIndex
+	}
+
+	call CopyReadGroupsToHeader {
+		input:
+			bam_with_readgroups = STAR.aligned_bam,
+			bam_without_readgroups = STAR.transcriptome_bam
+	}
+
+	call UmiMD.UMIAwareDuplicateMarking {
+		input:
+			aligned_bam = STAR.aligned_bam,
+			output_basename = output_basename
+	}
+
+	call UmiMD.UMIAwareDuplicateMarking as UMIAwareDuplicateMarkingTranscriptome {
+		input:
+			aligned_bam = CopyReadGroupsToHeader.output_bam,
+			output_basename = output_basename + ".transcriptome"
+	}
+
+	### PLACEHOLDER for CROSSCHECK ###
+
+	call GetSampleName {
+		input:
+			bam = bam
+	}
+
+	call rnaseqc2 {
+		input:
+			bam_file = UMIAwareDuplicateMarking.duplicate_marked_bam,
+			genes_gtf = gtf,
+			sample_id = GetSampleName.sample_name,
+			exon_bed = exonBedFile
+	}
+
+	call CollectRNASeqMetrics {
+		input:
+			input_bam=UMIAwareDuplicateMarking.duplicate_marked_bam,
+			input_bam_index=UMIAwareDuplicateMarking.duplicate_marked_bam_index,
+			output_bam_prefix=GetSampleName.sample_name,
+			ref_dict=refDict,
+			ref_fasta=ref,
+			ref_fasta_index=refIndex,
+			ref_flat=refFlat,
+			ribosomal_intervals=ribosomalIntervals,
+			preemptible_tries=0
+	}
+
+	call CollectMultipleMetrics {
+		input:
+			input_bam=UMIAwareDuplicateMarking.duplicate_marked_bam,
+			input_bam_index=UMIAwareDuplicateMarking.duplicate_marked_bam_index,
+			output_bam_prefix=GetSampleName.sample_name,
+			ref_dict=refDict,
+			ref_fasta=ref,
+			ref_fasta_index=refIndex,
+			preemptible_tries=0
+	}
+
+
+	output {
+		File transcriptome_bam = UMIAwareDuplicateMarkingTranscriptome.duplicate_marked_bam
+		File transcriptome_bam_index = UMIAwareDuplicateMarkingTranscriptome.duplicate_marked_bam_index
+		File transcriptome_duplicate_metrics = UMIAwareDuplicateMarkingTranscriptome.duplicate_metrics
+		File output_bam = UMIAwareDuplicateMarking.duplicate_marked_bam
+		File output_bam_index = UMIAwareDuplicateMarking.duplicate_marked_bam_index
+		File duplicate_metrics = UMIAwareDuplicateMarking.duplicate_metrics
+		File gene_tpm = rnaseqc2.gene_tpm
+		File gene_counts = rnaseqc2.gene_counts
+		File exon_counts = rnaseqc2.exon_counts
+		File metrics = rnaseqc2.metrics
```

Comment:
can we add `rnaseqc2.insert_size_histogram` as an output as well?  since this is also the name of a picard output we could either (a) give it a different name (I'd suggest fragment_size_histogram since that's what rnqseqc2 calls it) or we prefix all the rnaseqc2 metrics with `rnasecq2_`).

Reply:
Yes, it was added as `fragment_size_histogram`